torch/distributions/distribution.py

```diff
@@ -10,10 +10,17 @@ class Distribution(object):
 
     has_rsample = False
     has_enumerate_support = False
+    support = None
+    params = {}
 
-    def __init__(self, batch_shape=torch.Size(), event_shape=torch.Size()):
+    def __init__(self, batch_shape=torch.Size(), event_shape=torch.Size(), validate_args=False):
         self._batch_shape = batch_shape
         self._event_shape = event_shape
+        self._validate_args = validate_args
+        if self._validate_args:
+            for param, constraint in self.params.items():
+                if not constraint.check(self.__getattribute__(param)).all():
```

Comment:
I suspect that to get tests to pass, you'll need to add param attributes to some of the distributions, as pointed out by @tbrx. You could copy the `TestValid.test_valid` test from @lazypanda1's #134 to ensure this passes.

Reply:
All tests pass currently, without adding this. Is it still required to add `TestValid.test_valid`?

Reply:
Sure, I will cherry-pick it from the other branch.